app/containers/WeatherContainer/selectors.js

```diff
@@ -0,0 +1,34 @@
+import { createSelector } from 'reselect';
+import { initialState } from './reducer';
+import get from 'lodash/get';
+
+/**
+ * Direct selector to the weatherContainer state domain
+ */
+
+const selectWeatherContainerDomain = state => state.weatherContainer || initialState;
+
+export const selectWeatherContainer = () =>
+  createSelector(
+    selectWeatherContainerDomain,
+    substate => substate
+  );
+
+export const selectCityName = () =>
+  createSelector(
+    selectWeatherContainerDomain,
+    substate => get(substate, 'cityName', null)
+  );
+export const selectCityData = () =>
+  createSelector(
+    selectWeatherContainerDomain,
+    substate => get(substate, 'cityData', null)
```

Comment:
if we are defaulting this to null, in the reducer in failure cases lets make cityData as null too, currently its cityData= {}